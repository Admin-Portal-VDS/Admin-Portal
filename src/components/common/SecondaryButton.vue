<template>
  <div>
    <button class="secondary-button">
      {{ secondaryButton }}
    </button>
  </div>
</template>

<script>
export default {
  name: 'SecondaryButton',
  props: {
    secondaryButton: {
      type: String,
      required: true,
    },
  },
  created() {
    if (!this.secondaryButton) {
      throw new Error('Label prop is required');
    }
  },
};
</script>

<style scoped>
.secondary-button {
  border: 1.3px solid;
  border-radius: 6px;
  display: inline-block;
  font-size: 1rem;
  font-weight: 590;
  font-style: normal;
  text-align: center;
  padding: 8px 12px;
  min-height: 15px;
  min-width: 60px;
  color: #ffffff;
  background-color: #070000;
  appearance: none;
  margin-right: 500px;
}
</style>
