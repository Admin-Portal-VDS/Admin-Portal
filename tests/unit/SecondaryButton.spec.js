import { mount } from '@vue/test-utils';
import SecondaryButton from '@/components/common/SecondaryButton';

describe('SecondaryButton.vue', () => {
  it('renders button name correctly', () => {
    const secondaryButton = 'Secondary Button';
    const wrapper = mount(SecondaryButton, {
      propsData: {
        secondaryButton,
      },
    });
    const title = wrapper.find('button');
    expect(title.exists()).toBe(true);
    expect(title.text()).toBe(secondaryButton);
  });
  it('accepts props and passes them correctly', () => {
    const secondaryButton = 'Test Button';
    const wrapper = mount(SecondaryButton, {
      propsData: { secondaryButton },
    });
    expect(wrapper.props('secondaryButton')).toBe(secondaryButton); // Check prop
  });
  it('throws an error if required prop is missing', () => {
    expect(() => {
      mount(SecondaryButton);
    }).toThrow('Label prop is required');
  });
  it('matches snapshot', () => {
    const wrapper = mount(SecondaryButton, {
      propsData: { secondaryButton: 'Snapshot Test' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
