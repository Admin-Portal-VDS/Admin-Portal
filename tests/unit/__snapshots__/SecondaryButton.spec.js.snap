// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton.vue matches snapshot 1`] = `<div><button class="secondary-button">Snapshot Test</button></div>`;
